tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,214 @@
+            throw e;
+        } finally {
+            client.close();
+        }
+    }
+
+    private void seekToEndIntermediateTopics(final ZkUtils zkUtils, final List<String> allTopics) {
+        if (this.options.has(intermediateTopicsOption)) {
+            System.out.println("Seek-to-end for intermediate user topics.");
+
+            final Properties config = new Properties();
+            config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, this.options.valueOf(bootstrapServerOption));
+            config.setProperty(ConsumerConfig.GROUP_ID_CONFIG, this.options.valueOf(applicationIdOption));
+            config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+            for (final String topic : this.options.valuesOf(intermediateTopicsOption)) {
+                if (allTopics.contains(topic)) {
+                    System.out.println("Topic: " + topic);
+
+                    final KafkaConsumer<byte[], byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+                    try {
+                        client.subscribe(Collections.singleton(topic));
+                        client.poll(1);
+
+                        final Set<TopicPartition> partitions = client.assignment();
+                        client.seekToEnd(partitions);
+                        client.commitSync();
+                    } catch (final RuntimeException e) {
+                        System.err.println("Seek to end for topic " + topic + " failed");
+                        throw e;
+                    } finally {
+                        client.close();
+                    }
+                }
+            }
+
+            System.out.println("Done.");
+        } else {
+            System.out.println("No intermediate topics specified.");
+        }
+    }
+
+    private void deleteInternalTopics(final ZkUtils zkUtils, final List<String> allTopics) {
+        System.out.println("Deleting internal topics.");
+
+        for (final String topic : allTopics) {
+            if (isInternalStreamsTopic(topic)) {
+                final TopicCommand.TopicCommandOptions commandOptions = new TopicCommand.TopicCommandOptions(new String[]{
+                    "--zookeeper", this.options.valueOf(zookeeperOption),
+                    "--delete", "--topic", topic});
+                try {
+                    TopicCommand.deleteTopic(zkUtils, commandOptions);
+                } catch (final RuntimeException e) {
+                    System.err.println("Deleting internal topic " + topic + " failed");
+                    throw e;
+                }
+            }
+        }
+
+        System.out.println("Done.");
+    }
+
+    private boolean isInternalStreamsTopic(final String topicName) {
+        return topicName.startsWith(this.options.valueOf(applicationIdOption) + "-")
+            && (topicName.endsWith("-changelog") || topicName.endsWith("-repartition"));
```

Comment:
We would need a dependency from `kafka-tools` to `kafka-streams` -- I would rather not add this dependency.